x-pack/platform/plugins/shared/alerting/server/alerts_client/runtime_fields_manager.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty, get } from 'lodash';
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+import { MappingRuntimeField, MappingRuntimeFieldType } from '@elastic/elasticsearch/lib/api/types';
+import { IIndexPatternString } from '../alerts_service/resource_installer_utils';
+import { ReportedAlert } from './types';
+import {
+  AlertInstanceContext,
+  AlertInstanceState,
+  RuleAlertData,
+  WithoutReservedActionGroups,
+} from '../types';
+
+export class RuntimeFieldManager {
+  private logger: Logger;
+  private clientPromise: Promise<ElasticsearchClient>;
+  private readonly indexTemplateAndPattern: IIndexPatternString;
+  private readonly MAX_RUNTIME_FIELDS = 25;
+
+  private runtimeFields: Record<string, MappingRuntimeField> = {};
+
+  constructor({
+    indexTemplateAndPattern,
+    logger,
+    clientPromise,
+  }: {
+    indexTemplateAndPattern: IIndexPatternString;
+    logger: Logger;
+    clientPromise: Promise<ElasticsearchClient>;
+  }) {
+    this.indexTemplateAndPattern = indexTemplateAndPattern;
+    this.logger = logger;
+    this.clientPromise = clientPromise;
+  }
+
+  public async persistRuntimeFields() {
+    const { clientPromise, logger, MAX_RUNTIME_FIELDS, indexTemplateAndPattern } = this;
+
+    if (!isEmpty(this.runtimeFields)) {
+      const client = await clientPromise;
+
+      const response = await client.cluster.getComponentTemplate({
+        name: indexTemplateAndPattern.componentTemplate,
+      });
```

Comment:
Not necessary for PoC but when it comes to implementing, we should have some cache before queueing the task to add the runtime fields, so we don't queue tasks when we know the field already exists.